File: swagger-codegen/csharp-dotnet2/src/main/CsharpDotNet2/samsara/Model/VehicleHarshEventResponseLocation.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace samsara.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class VehicleHarshEventResponseLocation {
    /// <summary>
    /// Address of location where the harsh event occurred
    /// </summary>
    /// <value>Address of location where the harsh event occurred</value>
    [DataMember(Name="address", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "address")]
    public string Address { get; set; }

    /// <summary>
    /// Latitude of location where the harsh event occurred
    /// </summary>
    /// <value>Latitude of location where the harsh event occurred</value>
    [DataMember(Name="latitude", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "latitude")]
    public string Latitude { get; set; }

    /// <summary>
    /// Longitude of location where the harsh event occurred
    /// </summary>
    /// <value>Longitude of location where the harsh event occurred</value>
    [DataMember(Name="longitude", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "longitude")]
    public string Longitude { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class VehicleHarshEventResponseLocation {\n");
      sb.Append("  Address: ").Append(Address).Append("\n");
      sb.Append("  Latitude: ").Append(Latitude).Append("\n");
      sb.Append("  Longitude: ").Append(Longitude).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: openapi-generator/dart/lib/model/vehicle_harsh_event_response_location.dart
part of openapi.api;

class VehicleHarshEventResponseLocation {
  /* Address of location where the harsh event occurred */
  String address = null;
  /* Latitude of location where the harsh event occurred */
  String latitude = null;
  /* Longitude of location where the harsh event occurred */
  String longitude = null;
  VehicleHarshEventResponseLocation();

  @override
  String toString() {
    return 'VehicleHarshEventResponseLocation[address=$address, latitude=$latitude, longitude=$longitude, ]';
  }

  VehicleHarshEventResponseLocation.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['address'] == null) {
      address = null;
    } else {
          address = json['address'];
    }
    if (json['latitude'] == null) {
      latitude = null;
    } else {
          latitude = json['latitude'];
    }
    if (json['longitude'] == null) {
      longitude = null;
    } else {
          longitude = json['longitude'];
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'address': address,
      'latitude': latitude,
      'longitude': longitude
    };
  }

  static List<VehicleHarshEventResponseLocation> listFromJson(List<dynamic> json) {
    return json == null ? new List<VehicleHarshEventResponseLocation>() : json.map((value) => new VehicleHarshEventResponseLocation.fromJson(value)).toList();
  }

  static Map<String, VehicleHarshEventResponseLocation> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, VehicleHarshEventResponseLocation>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new VehicleHarshEventResponseLocation.fromJson(value));
    }
    return map;
  }
}

